internal/rbd/controllerserver.go

```diff
@@ -894,6 +899,11 @@ func (cs *ControllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteS
 			klog.Errorf(util.Log(ctx, "failed to delete rbd image: %s/%s with error: %v"), rbdVol.Pool, rbdVol.VolName, err)
 			return nil, status.Error(codes.Internal, err.Error())
 		}
+		err = rbdVol.removeImageFromTrash()
```

Comment:
Same here, removing from trash is probably not a real failure (from the CSI specs point of view)